src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/integration/AddressResourceIntTest.kt

```diff
@@ -634,6 +637,79 @@ class AddressResourceIntTest : SqsIntegrationTestBase() {
     private fun anUpdateCaAdditionalInfoRequest() = UpdateCaseAdminAdditionInfoRequest("some information")
   }
 
+  @Nested
+  inner class AddressDeleteReasonTests {
+    @Test
+    fun `should return unauthorized if no token`() {
+      webTestClient.post()
+        .uri(ADDRESS_DELETE_REASON)
+        .bodyValue(anAddressDeleteReasonRequest())
+        .exchange()
+        .expectStatus()
+        .isUnauthorized
+    }
+
+    @Test
+    fun `should return forbidden if no role`() {
+      webTestClient.post()
+        .uri(ADDRESS_DELETE_REASON)
+        .headers(setAuthorisation())
+        .bodyValue(anAddressDeleteReasonRequest())
+        .exchange()
+        .expectStatus()
+        .isForbidden
+    }
+
+    @Test
+    fun `should return forbidden if wrong role`() {
+      webTestClient.post()
+        .uri(ADDRESS_DELETE_REASON)
+        .headers(setAuthorisation(roles = listOf("ROLE_WRONG")))
+        .bodyValue(anAddressDeleteReasonRequest())
+        .exchange()
+        .expectStatus()
+        .isForbidden
+    }
+
+    @Sql(
+      "classpath:test_data/reset.sql",
+      "classpath:test_data/a-standard-address-check-request.sql",
+    )
+    @Test
+    fun `should update curfew address with address deletion reason`() {
+      webTestClient.post()
+        .uri(ADDRESS_DELETE_REASON)
+        .headers(setAuthorisation(roles = listOf("ASSESS_FOR_EARLY_RELEASE_ADMIN"), agent = PRISON_CA_AGENT))
+        .bodyValue(anAddressDeleteReasonRequest())
+        .exchange()
+        .expectStatus()
+        .isCreated
+
+      val addressCheckRequest = curfewAddressCheckRequestRepository.findByIdOrNull(ADDRESS_REQUEST_ID)
+      val addressDeletionEvent = addressDeletionEventRepository.findByIdOrNull(addressCheckRequest?.addressDeletionEvent?.id)
+      assertThat(addressCheckRequest?.addressDeletionEvent).isEqualTo(addressDeletionEvent)
```

Comment:
think it'd be better to explicitly test the specific fields have been set on the version set on `addressCheckRequest` rather than make two repository calls.

Reply:
+1